library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -135,7 +139,11 @@ private void deleteFrom(SqlDatabaseWrapper database, List<Migration> migrations)
         for (Migration migration : migrations) {
             Batch batch = migration.batch();
             database.rawQuery(DELETE_BY_ID_QUERY, batch.getDownloadBatchId().stringValue());
+            for (Migration.FileMetadata metadata : migration.getFileMetadata()) {
```

Comment:
Fix for bug two, will need further testing and the TODO addressing in subsequent PRs.